Running with pytest-xdist changes sys.path

A simple test to show sys.path (and then fail, to always see the output):
```
$ cat test_sys_path.py
import pprint, sys

def test_sys_path():
    pprint.pprint(sys.path)
    1/0
$ pip freeze
apipkg==1.5
atomicwrites==1.2.1
attrs==18.2.0
execnet==1.5.0
more-itertools==4.3.0
pluggy==0.8.0
py==1.7.0
pytest==3.10.1
pytest-forked==0.2
pytest-xdist==1.24.1
six==1.11.0
$ pytest -n 0 -q
F                                                                                                                                                        [100%]
=========================================================================== FAILURES ===========================================================================
________________________________________________________________________ test_sys_path _________________________________________________________________________

    def test_sys_path():
        pprint.pprint(sys.path)
>       1/0
E       ZeroDivisionError: division by zero

test_sys_path.py:5: ZeroDivisionError
--------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------
['/private/var/folders/j2/gr3cj3jn63s5q8g3bjvw57hm0000gp/T/pytest_sys_path',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/bin',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python36.zip',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6/lib-dynload',
 '/usr/local/py/cpython-v3.6.7/lib/python3.6',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6/site-packages']
1 failed in 0.05 seconds
$ pytest -n 1 -q
gw0 [1]
scheduling tests via LoadScheduling
F                                                                                                                                                        [100%]
=========================================================================== FAILURES ===========================================================================
________________________________________________________________________ test_sys_path _________________________________________________________________________
[gw0] darwin -- Python 3.6.7 /usr/local/virtualenvs/tmp-9cf2e53a94c913c/bin/python3.6

    def test_sys_path():
        pprint.pprint(sys.path)
>       1/0
E       ZeroDivisionError: division by zero

test_sys_path.py:5: ZeroDivisionError
--------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------
['/private/var/folders/j2/gr3cj3jn63s5q8g3bjvw57hm0000gp/T/pytest_sys_path',
 '',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python36.zip',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6/lib-dynload',
 '/usr/local/py/cpython-v3.6.7/lib/python3.6',
 '/usr/local/virtualenvs/tmp-9cf2e53a94c913c/lib/python3.6/site-packages']
1 failed in 0.54 seconds
```
The second entry in sys.path is different: without xdist, it's the bin directory of the virtualenv.  With xdist, it's an empty string (current directory).  

xdist shouldn't change this, it means my tests behave differently under the two scenarios.

--- a/changelog/376.feature.rst
+++ b/changelog/376.feature.rst
@@ -1,0 +1,3 @@
+The current directory is no longer added ``sys.path`` for local workers, only for remote connections.
+
+This behavior is surprising because it makes xdist runs and non-xdist runs to potentially behave differently.

--- a/xdist/remote.py
+++ b/xdist/remote.py
@@ -251,17 +251,20 @@ def remote_initconfig(option_dict, args):
 
 
 if __name__ == "__channelexec__":
+    import py
+
     channel = channel  # noqa
-    workerinput, args, option_dict = channel.receive()
-    importpath = os.getcwd()
-    sys.path.insert(0, importpath)  # XXX only for remote situations
-    os.environ["PYTHONPATH"] = (
-        importpath + os.pathsep + os.environ.get("PYTHONPATH", "")
-    )
+    workerinput, args, option_dict, change_sys_path = channel.receive()
+
+    if change_sys_path:
+        importpath = os.getcwd()
+        sys.path.insert(0, importpath)
+        os.environ["PYTHONPATH"] = (
+            importpath + os.pathsep + os.environ.get("PYTHONPATH", "")
+        )
+
     os.environ["PYTEST_XDIST_WORKER"] = workerinput["workerid"]
     os.environ["PYTEST_XDIST_WORKER_COUNT"] = str(workerinput["workercount"])
-    # os.environ['PYTHONPATH'] = importpath
-    import py
 
     config = remote_initconfig(option_dict, args)
     config._parser.prog = os.path.basename(workerinput["mainargv"][0])

--- a/xdist/workermanage.py
+++ b/xdist/workermanage.py
@@ -245,7 +245,9 @@ class WorkerController(object):
                 option_dict["basetemp"] = str(basetemp.join(name))
         self.config.hook.pytest_configure_node(node=self)
         self.channel = self.gateway.remote_exec(xdist.remote)
-        self.channel.send((self.workerinput, args, option_dict))
+        # change sys.path only for remote workers
+        change_sys_path = not self.gateway.spec.popen
+        self.channel.send((self.workerinput, args, option_dict, change_sys_path))
         if self.putevent:
             self.channel.setcallback(self.process_from_remote, endmarker=self.ENDMARK)
 

